includes/class-search-categories-metabox.php

```diff
@@ -0,0 +1,281 @@
+<?php
+
+class Search_Categories_Metabox {
+
+	/**
+	 * @const PLUGIN_SLUG
+	 */
+	CONST PLUGIN_SLUG = 'search-categories-metabox';
+
+	/**
+	 * @var string $plugin_file
+	 */
+	protected $plugin_file;
+
+	/**
+	 * @var string $plugin_version
+	 */
+	public $plugin_version = '0.1.0';
+
+	/**
+	 * Search_Categories_Metabox constructor.
+	 * @param $file
+	 */
+	public function __construct( $file ) {
+		$this->plugin_file = $file;
+	}
+
+	/**
+	 * Gets the Post taxonomy Category slug by Post ID.
+	 *
+	 * @since 0.1.0
+	 * @access private
+	 *
+	 * @param $post_id
+	 * @return array
+	 */
+	private function get_post_taxonomy( $post_id ) {
+		$taxonomies = get_post_taxonomies( $post_id );
+		$taxonomy   = $taxonomies[0];
+		return $taxonomy;
+	}
+
+	/**
+	 * Saves the Post Categories to DB.
+	 *
+	 * @since 0.1.0
+	 */
+	public function save_post_categories() {
+		global $post;
+
+		$post_taxonomy       = $this->get_post_taxonomy( $post->ID );
+		$selected_categories = $_POST['post-category'];
+		if ( isset( $selected_categories ) && ! empty( $selected_categories ) ) {
+			$selected_categories = array_map( 'intval', $selected_categories );
+			$selected_categories = array_unique( $selected_categories );
+			wp_set_object_terms( $post->ID, $selected_categories, $post_taxonomy, false );
+		}
+	}
+
+	/**
+	 * Gets the array of categories.
+	 *
+	 * @since 0.1.0
+	 * @access private
+	 *
+	 * @param string $search_term
+	 * @return array
+	 */
+	protected function get_taxonomy_categories( $search_term = '' ) {
+		$categories = array();
+		$args       = array();
+		$taxonomy   = array();
+
+		/*
+		 * Get only Category taxonomy
+		 * Refer https://codex.wordpress.org/Function_Reference/get_post_taxonomies
+		 */
+		$taxonomy   = $this->get_post_taxonomy();
+
+		$args = array(
+			'hide_empty' => false,
+			'order_by'   => 'term_id',
+			'order'      => 'ASC',
+			'taxonomy'   => $taxonomy,
+		);
+
+		if ( trim( $search_term ) != '' ) {
+
+			$additional_args = array(
+				'name__like' => $search_term,
+				'number'     => 10,
+			);
+
+			$args = wp_parse_args( $additional_args, $args );
+		}
+
+		$categories = get_categories( $args );
```

Comment:
You can directly return it instead of storing it in a variable and then returning it.